featuretools/tests/synthesis/test_dask_dfs.py

```diff
@@ -54,7 +54,9 @@ def test_single_table_dask_entityset():
                    trans_primitives=primitives_list)
 
     # Use the same columns and make sure both indexes are sorted the same
-    dask_computed_fm = dask_fm.compute().set_index('id').loc[fm.index][fm.columns]
+    # update the type of the future index column so it doesn't conflict with the pandas fm
+    dask_fm = dask_fm.compute().astype({'id': 'int64'})
+    dask_computed_fm = dask_fm.set_index('id').loc[fm.index][fm.columns]
```

Comment:
Most of the failing tests were because of a changed (fixed, actually, I think) behavior in `set_index`. Previously, the following would result in the index's dtype no longer being `Int64`, and now it's retained as `Int64`, but that means it now will get caught in the situation that came from #1810, so updating the dtype to be `int64` to begin with fixes this problem.

```python
 df = pd.DataFrame({
     'a': pd.Series([1, 2, 3,4], dtype='int64'),
     'b': pd.Series([1, 2, 3,3], dtype='Int64'),
 })
df.set_index('b')
```

Reply:
Are we able to revert some of the `check_dtype=False` changes that were introduced in #1810?

Reply:
There are columns beyond `id` that will have different dtypes from the pandas dataframe (`values`, for example, which is Integer for Pandas and IntegerNullable otherwise because of the changes in #1810 ). So we need to keep it in